lib/src/rules/unnecessary_await.dart

```diff
@@ -0,0 +1,97 @@
+// Copyright (c) 2018, the Dart project authors. Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'package:analyzer/dart/ast/ast.dart';
+import 'package:analyzer/dart/ast/visitor.dart';
+import 'package:analyzer/dart/element/type.dart';
+import 'package:linter/src/analyzer.dart';
+
+const _desc = r'Unnecessary await keyword.';
+
+const _details = r'''
+
+Avoid using await on an expression with a type that is assignable to the return
+type.
```

Comment:
This seems awkward though I admit I'm struggling to improve it.

Maybe something like:

    Avoid returning an awaited expression when the expression type is assignable to a function's return type.

Likely @bwilkerson can do better...

Reply:
Changed with Phil suggestion.